docs/docs/life-and-times-of-a-gatsby-build.md

```diff
@@ -2,7 +2,311 @@
 title: Life and times of a Gatsby build
 ---
 
-This is a stub. Help our community expand it.
+import LayerModel from "../../www/src/components/layer-model"
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+<!-- written at the time of commit e85278c to bootstrap/index.js and commit f8cae16 to build.js -->
+
+_This is a high-level overview about the steps in the Gatsby build process. For more detailed information about specific steps you can find information in the [Gatsby Internals](/docs/gatsby-internals) section of the docs._
+
+Gatsby has two modes for compiling a site:
+
+1. Develop - run with the `gatsby develop` command
+2. Build - run with `gatsby build`
+
+You can start Gatsby in either mode with its respective command: `gatsby develop` or `gatsby build`.
+
+## Build time vs runtime
+
+A common confusion with the generation of [static](/docs/glossary#static) assets is the difference between [build time](/docs/glossary#build) and [runtime](/docs/glossary#runtime).
+
+Processes that happen in a web browser that you click through and interact with can be referred to as a _browser runtime_. JavaScript code can interact with the browser and take advantage of APIs it offers, such as `window.location` or injecting markup dynamically into the [DOM](/docs/glossary#dom). [Gatsby creates a JavaScript runtime](/docs/production-app) that takes over in the browser once the initial HTML has loaded.
+
+_Build time_, in contrast, refers to the process of using a server process to compile the site into files that can be delivered to a web browser later, so DOM APIs like `window` aren't available at that time.
```

Comment:
Technically it's browser apis as DOM apis only refer to things like querySelector, .... Browser APIs are functions like fetch, serviceworker, ....
```suggestion
_Build time_, in contrast, refers to the process of using a server process to compile the site into files that can be delivered to a web browser later, so Browser APIs like `window` aren't available at that time.
```